libcxx/include/istream

```diff
@@ -1211,12 +1211,12 @@ operator>>(basic_istream<_CharT, _Traits>& __is, basic_string<_CharT, _Traits, _
     try {
 #endif
       __str.clear();
-      streamsize __n = __is.width();
-      if (__n <= 0)
-        __n = __str.max_size();
-      if (__n <= 0)
-        __n = numeric_limits<streamsize>::max();
-      streamsize __c            = 0;
+      using _Size = typename basic_string<_CharT, _Traits, _Allocator>::size_type;
+      static_assert(numeric_limits<_Size>::max() >= numeric_limits<streamsize>::max(),
```

Comment:
Why...? Can't set allocator size_type to 32-bit?

Reply:
The other alternative would be to use `streamsize` for the counter and do something like:

```
streamsize const __width = __is.width();
streamsize const __n = __width <= 0 ? std::clamp(__str.max_size(), 0, numeric_limits<streamsize>::max()) : __width;
```

WDYT?

Reply:
`__str.max_size()` may not be representable in `streamsize`, either. And we'd better not let `__n` exceed `max_size()` in practice, so it looks reasonable to count by `size_type`. If `__is.width()` is larger than `max_size()` at runtime, stores `max_size()`.

Reply:
Let me know what you think about the latest version.